### PR TITLE
251 use jinja for job script rendering

### PR DESCRIPTION
**Pull request by bschroeter**
Replaced in-source job script generation with Jinja templating.

For context - this simplifies the generation of job scripts (and indeed anything that interpolates into a template), meaning that we can remove the multiline string blocks, which are sensitive to whitespace, and abstract them into separate files.

This results in cleaner and more flexible code going forward and opens up opportunities to change the underlying templates at a moments notice.

### CODE REVIEW

File: ruff.toml [hunk 1]
```diff
@@ -6,4 +6,5 @@ ignore = [
     "D104",  # D104 Missing docstring in public package
     "PLR0913", # PLR0913 Too many arguments in function definition
     "D100", # D100 Missing docstring in public module
+    "I001", # Import block is un-sorted or un-formatted
```
Comment (bschroeter) [inline ruff.toml:9]: This appears to be a necessary evil - the license headers are triggering the sorted imports rule.

File: benchcab/data/pbs_jobscript.j2 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#PBS -l mem={{mem}}
+#PBS -l walltime={{walltime}}
+#PBS -q normal
+#PBS -P {{project}}
+#PBS -j oe
+#PBS -m e
+#PBS -l storage={{storage}}
+
+module purge
+{% for module in modules -%}
```
Comment (bschroeter) [inline benchcab/data/pbs_jobscript.j2:13]: Note the -%, which controls whitespace generation in Jinja. 

https://jinja.palletsprojects.com/en/latest/templates/#whitespace-control

**Comment by bschroeter** [comment]
All tests pass btw.

```
PBS Job Id: 108301484.gadi-pbs
Job Name:   benchmark_cable_qsub.sh
Execution terminated
Exit_status=0
resources_used.cpupercent=371
resources_used.cput=00:05:21
resources_used.jobfs=0b
resources_used.mem=601672kb
resources_used.ncpus=18
resources_used.vmem=601672kb
resources_used.walltime=00:01:39
```

File: ruff.toml (hunk 1 above)
Comment (ccarouge) [reply ruff.toml:9]: That's weird. It works fine with the headers beforehand...

**Review by ccarouge — APPROVED** [review]
A small comment about typing. Otherwise all good. I approve but it would be good to fix the typing before merging if I am correct in my comment.

File: benchcab/utils/__init__.py [hunk 3]
```diff
@@ -48,10 +48,54 @@ def load_package_data(filename: str) -> dict:
...
+        Keyword arguments to interpolate into the file.
+
+    Returns
+    -------
+    str
+        Interpolated template string.
+
+    """
+    template = load_package_data(template_file)
+    return interpolate_string_template(template, **kwargs)
```
[the remark this answers is not in the document]
Comment (bschroeter) [reply benchcab/utils/__init__.py:97]: yes you are correct, now that we are potentially reading raw data this is needed. I'll update that now.

**Pull request merged**